### DISCUSSION TITLE
Identify Signatures on separate lines of readout

### DISCUSSION THREAD

**Discussion by Posvar** [ImageMagick/ImageMagick#3919] [Help]
I'm tyring to produce the 64 digit signatures for a large (9K) batch of PNG files:

`magick identify -quiet -format "%#" *.png > Signature.txt`

Let's say I had 2 PNGs in the folder - I'd want the output to have two lines of text
Line1: 64 character signature
Line2: 64 character signature

Instead - running the above just creates a single line of concatenated text - all 128 characters

Is there a way to tell IM to output each image's signature to a separate line - even better would be to output the image name on the same row:

image1.png     64 character signature
image2.png     64 character signature

**Comment by snibgo** [comment]
For escapes, such as %#, see https://imagemagick.org/script/escape.php .

"%f" is the filename. You can use other characters, such as "\n" for a newline. So:

```
f:\pictures\20210623>%IMG7%magick identify -quiet -format "%f %#\n" b*.png

b.png 4dcd4edf0616acb981f8e65a1ac309064d91db6740b493afaf67b3fb46ed0a68
b2.png 5259ecd5f7b2864cf836c8736f113f81b6fda843ff5f44767d029e64ff760138
b3.png 5f11c60b89fe7a87fc4777ffc98b0824665e638cc329417a095cf076e93c0ca7
b4.png 01437c9d7a8da2e3bb46d1f0ace59235a766146c4609f7ca6aa5c11bc8577ee6
b5.png 4e1652245bfedacc2eb1c1eea4234aeb469211e11e1cb210ec1eea0833ec5f98
b6.png 9b2220d908134425e484a29a2e53cdbcc15c88e47929a82bf21469374cd443b3
```

**Reply by Posvar** [reply]
OMG what a legend - thanks snibgo - been trying to figure this out for way tooooo long.  Appreciate you.
